feat(bundle,source-inclusion): allow `exclude` option in bundle source

AStoker opened the pull request:
Fixes: https://github.com/aurelia/cli/issues/238
Address some of the issues mentioned here: https://github.com/aurelia/cli/issues/203  
Allow the `source` option in the bundle config to optionally be an
object with two properties, `include`, and `exclude`. This makes it
easier for patterns to be defined by the user, and allows for a more
consistent approach to creating bundles (similar to the aurelia-bundler)

EisenbergEffect commented on lib/build/source-inclusion.js:
```diff
@@ -27,8 +28,16 @@ exports.SourceInclusion = class {
 
+  isExcluded(item){
+	  let found = this.excludes.findIndex(x => {			
+			let exclusion = new this.bundle.Minimatch(x);
```
I'm mainly wondering if this could be more efficient by creating the Minimatch objects once, upfront, rather than on every call to isExcluded. @AStoker Any thoughts on that?

AStoker replied:
I was thinking about that, but wasn't set on exactly where to place it. I gotta drive home from work early to avoid the flash floods here, so I'll find a more efficient place for it and have it in in an hour or so.

EisenbergEffect replied:
Sounds good.







> On Sep 2, 2016, at 3:21 PM, Andrew Stoker <notifications@github.com> wrote:
> 
> 
> In lib/build/source-inclusion.js
> :
> 
>> @@ -27,8 +28,16 @@ exports.SourceInclusion = class {
>>      this.items.push(item);
>>    }
>>  
>> +  isExcluded(item){
>> +	  let found = this.excludes.findIndex(x => {			
>> +			let exclusion = new this.bundle.Minimatch(x);
> I was thinking about that, but wasn't set on exactly where to place it. I gotta drive home from work early to avoid the flash floods here, so I'll find a more efficient place for it and have it in in an hour or so.
> 
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> , or mute the thread
> .<embedded image>
> 
>

AStoker replied:
Part of it might be my lack of knowledge of Minimatch, but I'm not entirely sure how the best way to go about creating the matcher up front. Where the `include` matcher can have a single pattern and loop through an array of files, the `exclude` matcher needs to have a file loop through an array of `exclusions`.  
Perhaps it's the Friday in me. I think I might be able to create the exclusion list on the bundle itself (instead of the source-inclusion), then each exclusion is just created once, and when the source-inclusion goes through trying to match files, it'll match against the bundle's exclusion list. How does that approach sound @EisenbergEffect?

EisenbergEffect replied:
Yes, that's basically the approach I was thinking of. Give it a shot and if it works, amend this PR and I'll get it merged and released this weekend.

AStoker commented:
@EisenbergEffect, moved the exclusion list creation to here: https://github.com/AStoker/cli/blob/source-exclusion/lib/build/bundle.js#L22
Let me know what you think. 
Also, I added some details to how the bundles work in the `README.md`. There weren't any details before, so I added some basic details to both how bundles work normally, and with the new syntax.

EisenbergEffect commented:
Release coming in a few minutes.
